### ISSUE TITLE
Clean up Amazon EBS Volumes created by Packer upon completion

### ISSUE THREAD

**Issue by RyanGWU82** [mitchellh/packer#1783]
I'm using the `amazon-ebs` builder to build an image based on the source AMI `ami-96a818fe`, which is [the official CentOS 7 AMI](https://aws.amazon.com/marketplace/pp/B00O7WM7QW). This AMI creates an 8GB root volume on EBS, and the `delete_on_termination` flag is `false`. (This issue should be reproducible on any AMI where the `delete_on_termination` flag is `false`, but it's definitely reproducible using this one.)

Packer itself runs fine and builds an AMI without any problems. But when Packer finishes, it does not delete the instance's root EBS volume, although it cleanly terminates the source instance and all the other resources it created.

Packer should delete this EBS volume too. It's not an artifact of the build process; it's just something left over from Packer's operation.

**Comment by pas256** [comment]
That makes sense.

An alternative to get around this is to use the `amazon-chroot` builder, which will not have this issue.

**Comment by ktheory** [comment]
@RyanGWU82 Have you tried overriding the `delete_on_termination` setting by passing `launch_block_device_mappings` to your builder? That let's you override the block-device mappings built in to the AMI at launch time.

E.g.
```
"launch_block_device_mappings": [{
  "device_name": "/dev/sda1",
  "delete_on_termination": true
}]
```

**Comment by RyanGWU82** [comment]
@ktheory Great, that did the trick. It would still be helpful if Packer did this automatically, but for now we can use the `launch_block_device_mappings` directive in our config.

**Comment by catsby** [comment]
I've opened a PR to patched this https://github.com/mitchellh/packer/pull/2282

**Comment by catsby** [comment]
Just merged #2282 , let me know if you have any questions!
